samples/cpp/benchmark_app/main.cpp

```diff
@@ -463,17 +527,20 @@ int main(int argc, char* argv[]) {
                 auto property_name = str == "nthreads" ? ov::inference_num_threads.name() : ov::affinity.name();
                 auto property = str == "nthreads" ? ov::inference_num_threads(FLAGS_nthreads)
                                                   : ov::affinity(fix_pin_option(FLAGS_pin));
-                if (supported(property_name) || if_multi) {
+                if (supported(property_name) || device_name == "AUTO") {
+                    // create nthreads/pin primary property for HW device or AUTO if -d is AUTO directly.
                     device_config.emplace(property);
-                } else if (if_auto) {
+                } else if (if_auto || if_multi) {
+                    // create nthreads/pin secondary property setting for each hw device from hw device list if -d
```

Comment:
I'm not sure I understand what happens here.

-nthreads and -pin are applicable only for CPU hardware device. But the comment is formulated like
> create nthreads/pin secondary property setting for each hw device...

That suggests there is more that one hardware device supporting -nthreads and -pin.

Reply:
You are right. Just create secondary proerty with -nthreads/-pin for device CPU. The logic here was to set -nthreads/-pin only for the device who supports these 2 properties.